test/stable_build_test.go

```diff
@@ -77,13 +77,14 @@ var _ = Describe("go build caching", func() {
 			return strings.TrimSpace(string(out))
 		}
 
-		checkCacheSize := func() string {
+		checkCacheFiles := func() string {
 			d := getCachedir()
 			GinkgoWriter.Println("Cache:", d)
-			// there's just one source file in the project, so the difference between the cache
-			// working and it not working will be pretty small. `-k` measures in 1KiB blocks, which
-			// should be fine enough.
-			out, err := exec.Command("du", "-s", "-k", d).Output()
+			// there's just one source file in the project, so the size difference between the cache
+			// working and it not working will be pretty small. However, there certainly should not
+			// be _new_ files after building again; so, listing all the files, should catch cache
+			// growth.
+			out, err := exec.Command("ls", "-aR", d).Output()
```

Comment:
Happy with shelling out to `ls`. But also prefer not to shell out when possible, so using`filepath.Walk` or `os.ReadDir`. I'll defer the choice to you.

Reply:
Yeah, I have the same preference, in general. I am OK with this instance because it's more or less a one-liner, and that invocation should be stable across OSes (ubiquitous command, and standard flags).